kbfshash/hash_test.go

```diff
@@ -226,3 +226,24 @@ func TestVerify(t *testing.T) {
 	err = corruptHMAC.Verify(key, data)
 	require.IsType(t, HashMismatchError{}, errors.Cause(err))
 }
+
+var defaultHashBenchmarkResult bool
+var defaultHashBenchmarkError error
+
+func BenchmarkDefaultHash(b *testing.B) {
+	data := make([]byte, b.N)
+	for i := 0; i < b.N; i++ {
+		data[i] = byte(i % 256)
+	}
+
+	isValid := false
+	var err error
+
+	for i := 0; i < b.N; i++ {
+		h, _ := DefaultHash(data)
+		isValid = h.IsValid()
+		err = h.Verify(data)
+	}
+	defaultHashBenchmarkResult = isValid
```

Comment:
What are these global `default` values good for?  Why not use the regular test failure/require methods?

Reply:
https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go#compiler-optimisation

Reply:
You can't use require because there is no `testing.T`, just `testing.B`.

Reply:
Oh crazy about the optimization thing.

Yeah I guess you can't use require, but you could still use `b.Error`, etc.